internal/wal/wal.go

```diff
@@ -4,19 +4,17 @@
 package wal
 
 import (
-	"fmt"
 	"log/slog"
 	sync "sync"
 	"time"
-
-	"github.com/dicedb/dice/internal/cmd"
 )
 
 type AbstractWAL interface {
 	LogCommand([]byte) error
 	Close() error
 	Init(t time.Time) error
-	ForEachCommand(f func(c cmd.DiceDBCmd) error) error
+	ReplayWAL(c func(*WALEntry) error) error
```

Comment:
The function should be called `Replay` and not `ReplayWAL` given we are already under type `AbstractWAL`